src/PageParser.php

```diff
@@ -294,21 +294,28 @@ protected function cleanIds(): void {
 			if ( array_search( $id, self::$ids ) !== false ) {
 				$id .= '-n' . ( array_count_values( self::$ids )[ $id ] + 1 );
 			}
+			// Transliterate ID and ensure it contains only certain characters.
+			$idTrans = transliterator_transliterate( 'Any-Latin; Latin-ASCII; ', $id );
+			$idTransConv = iconv( 'UTF-8', 'ASCII//TRANSLIT//IGNORE', $idTrans );
+			$id = preg_replace( '/[^a-zA-Z0-9\-_:.]/', '_', str_replace( ' ', '_', $idTransConv ) );
```

Comment:
I don't see why the `str_replace` is useful.

Reply:
Good point. Neither do I! :-) Removed.